dataflow/gpu-workers/tensorflow-landsat/main.py

```diff
@@ -276,39 +276,42 @@ def run(
     max_value = vis_params["max"]
     gamma = vis_params["gamma"]
 
-    options = PipelineOptions(beam_args, save_main_session=True)
-    with beam.Pipeline(options=options) as pipeline:
-        # Optionally, validate that the workers are using GPUs.
-        gpu_check = (
-            pipeline
-            | beam.Create([None])
-            | "Check GPU availability" >> beam.Map(check_gpus, gpus_optional)
+    beam_options = PipelineOptions(beam_args, save_main_session=True)
+
+    # We currently cannot use the `with` statement to run without waiting.
+    #   https://issues.apache.org/jira/browse/BEAM-12455
+    pipeline = beam.Pipeline(options=beam_options)
+
+    # Convert Landsat 8 scenes into images.
+    # ℹ️ We pass `gpu_check` as an unused side input to force that step in
+    # the pipeline to wait for the check before continuing.
```

Comment:
How abount:

We pass `gpu_check` as an unused side input to make a blocking one-time check
in the pipeline to verify GPUs are available.

Reply:
Removed this and used a more compact pattern.